### ISSUE TITLE
Confuse about dump "tessnoimages.png" when segmenting page and detect orientation. 

### ISSUE THREAD

**Issue by nam-leduc** [tesseract-ocr/tesseract#269]
I see in file "ccmain/pagesegmain.cpp: line 322 - 333". 

``` c++
  // Leptonica is used to find the rule/separator lines in the input.
  LineFinder::FindAndRemoveLines(source_resolution_,
                                 textord_tabfind_show_vlines, pix_binary_,
                                 &vertical_x, &vertical_y, music_mask_pix,
                                 &v_lines, &h_lines);
  if (tessedit_dump_pageseg_images)
    pixWrite("tessnolines.png", pix_binary_, IFF_PNG);
  // Leptonica is used to find a mask of the photo regions in the input.
  *photo_mask_pix = ImageFind::FindImages(pix_binary_);
  if (tessedit_dump_pageseg_images)
    pixWrite("tessnoimages.png", pix_binary_, IFF_PNG);
```

Please notice to line: 

``` c++
  if (tessedit_dump_pageseg_images)
    pixWrite("tessnoimages.png", pix_binary_, IFF_PNG);
```

I try with configuration:

```
tessedit_dump_pageseg_images    1
```

And images: 
![news](https://cloud.githubusercontent.com/assets/8704662/13735304/7b94f300-e9da-11e5-94f0-c4eeac8ca254.png)

But result in file "tessnoimages.png" is: 
![tessnoimages](https://cloud.githubusercontent.com/assets/8704662/13735340/bf23a256-e9da-11e5-97d1-ee3d6e1e9128.png)

I think before `pixWrite("tessnoimages.png", pix_binary_, IFF_PNG);`, we need something like:

``` c++
if (*photo_mask_pix != NULL) {
    pixSubtract(pix_binary_, pix_binary_, *photo_mask_pix);    
}
```

**Comment by zdenop** [comment]
At least in [osdetect](https://github.com/tesseract-ocr/tesseract/blob/master/ccmain/osdetect.cpp#L177) it is done that way. Anyway I found out that   pixSubtract(pix_binary_, pix_binary_, *photo_mask_pix) does not removed all images.
@theraysmith: can you have a look at this?

**Comment by stweil** [comment]
@nam-leduc, is this still an issue with latest Tesseract? We'd like to know that before releasing Tesseract 4.0.0.

**Comment by zdenop** [comment]
@stweil: I testing it with:

`tesseract i269.png i269 -c tessedit_dump_pageseg_images=1`

Interesting it that it produce pdf ([i269_debug.pdf](https://github.com/tesseract-ocr/tesseract/files/2418680/i269_debug.pdf)) for debugging steps and it seems it is not fixed...

**Comment by amitdo** [comment]
https://github.com/tesseract-ocr/tesseract/blob/641237495a4e8b07e3fc6110c694c4a9e3ee3dab/src/ccmain/pagesegmain.cpp#L294

https://github.com/tesseract-ocr/tesseract/blob/a9e2574effcbc589eba9d7ced92320cd920e79f8/src/textord/imagefind.cpp#L63

https://github.com/DanBloomberg/leptonica/blob/master/prog/livre_pageseg.c

**Comment by nam-leduc** [comment]
Dear @amitdo , 

Recently, I tested and it still meet the problems.

### Environment

- Tesseract Version:
  ```
  tesseract --version
  tesseract 5.0.0-alpha-777-g162f3
   leptonica-1.75.3
    libgif 5.1.4 : libjpeg 8d (libjpeg-turbo 1.5.2) : libpng 1.6.34 : libtiff 4.0.9 : zlib 1.2.11 : libwebp 0.6.1 : libopenjp2 2.3.0

   Found AVX2
   Found AVX
   Found SSE
  ```

- Commit: 162f3707e28102451338ced235c80ffa6f09ae40 (Merge pull request #3082 from bertsky/fix-line-detector)
- Platform: Ubuntu 18.04.4 LTS
- Tessdata: https://raw.githubusercontent.com/tesseract-ocr/tessdata/master/eng.traineddata

### Tesseract input/output
  - Following are input image:
  https://user-images.githubusercontent.com/8704662/92300832-517da180-ef88-11ea-8ce3-149bd3e79bc3.PNG

  - My command:
    ```
    $ ./bin/tesseract ~/vmshare/have-image.PNG have-image-original -c tessedit_dump_pageseg_images=1
    Tesseract Open Source OCR Engine v5.0.0-alpha-777-g162f3 with Leptonica
    ```

  - Following are PDF debug file when run with original source code: 
  [have-image-original_debug.pdf](https://github.com/tesseract-ocr/tesseract/files/5177746/have-image-original_debug.pdf)

  - And this is image after I add following source code:
    [have-image_debug.pdf](https://github.com/tesseract-ocr/tesseract/files/5177735/have-image_debug.pdf)
    ```c++
    *photo_mask_pix = ImageFind::FindImages(pix_binary_, &pixa_debug_);
    if (tessedit_dump_pageseg_images) {
      Pix* pix_no_image_ = nullptr;
      if (*photo_mask_pix != nullptr) {
        pix_no_image_ = pixSubtract(nullptr, pix_binary_, *photo_mask_pix);
      } else {
        pix_no_image_ = pixClone(pix_binary_);
      }
      pixa_debug_.AddPix(pix_no_image_, "NoImages");
      pixDestroy(&pix_no_image_);
    }
    ```

Best regards,
Le Duc. Nam

**Comment by amitdo** [comment]
LGTM.

**Comment by amitdo** [comment]
>Anyway I found out that pixSubtract(pix_binary_, pix_binary_, *photo_mask_pix) does not removed all images.

@zdenop,  

Can you give an example?

Probably Leptonica itself fails to identify the these images. For example, Leptonica does not recognize line drawings as images.

**Comment by zdenop** [comment]
@amitdo: as you can see in [have-image_debug.pdf](https://github.com/tesseract-ocr/tesseract/files/5177735/have-image_debug.pdf) from @nam-leduc example (or you can use original image from  this issue) - there are still image artefacts.

[Leduct patch](https://github.com/tesseract-ocr/tesseract/issues/269#issuecomment-687569986) improves debug output without effect on OCR (so it could/should be applied). @nam-leduc: can you send PR?

My original suggestion has effect on OCR and I did not test it  (so it is dangerous to apply it). IMHO [SetupAndFilterNoise ](https://github.com/tesseract-ocr/tesseract/blob/162f3707e28102451338ced235c80ffa6f09ae40/src/textord/colfind.cpp#L146 )should be investigated if `photo_mask_pix/FindImages` produces expected results.

**Comment by nam-leduc** [comment]
Hi @zdenop ,

> @nam-leduc: can you send PR?

Yes, I will send PR.

**Comment by stweil** [comment]
I suggest to simplify the patch a little bit. It is not necessary to write a 2nd image for `pix_binary_`, so the `else` part can be omitted..

**Comment by nam-leduc** [comment]
Hi @stweil , my thinking is that if we can't find any photo from binary image, so, binary image are 'NoImages', it maybe confuse developer if we don't add 'NoImages' pdf page. @stweil , what do you think?

**Comment by stweil** [comment]
Fixed by pull request #3090, so closing this issue.
